src/uu/du/src/du.rs

```diff
@@ -255,22 +253,27 @@ fn get_file_info(path: &Path) -> Option<FileInfo> {
     result
 }
 
-fn read_block_size(s: Option<&str>) -> u64 {
+fn read_block_size(s: Option<&str>) -> UResult<u64> {
     if let Some(s) = s {
-        parse_size_u64(s)
-            .unwrap_or_else(|e| crash!(1, "{}", format_error_message(&e, s, options::BLOCK_SIZE)))
+        match parse_size_u64(s) {
+            Ok(x) => Ok(x),
+            Err(e) => Err(USimpleError::new(
+                1,
+                format_error_message(&e, s, options::BLOCK_SIZE),
+            )),
+        }
```

Comment:
maybe simplify with:
```suggestion
parse_size_u64(s)
    .map_err(|e| USimpleError::new(
        1,
        format_error_message(&e, s, options::BLOCK_SIZE),
    ))
```